src/classes/TDTM_Config.cls

```diff
@@ -0,0 +1,153 @@
+    * @description Holds the trigger handlers cached in a list, to avoid repetitive soql calls.
+    *********************************************************************************************************/
+    private static List<TDTM_Global_API.TdtmToken> tdtmTokensCache;
+    
+    /*******************************************************************************************************
+    * @description Number of trigger handlers records that currently exist in the system.
+    *********************************************************************************************************/
+    private static Integer rowCount;
+    
+    /********************************************************************************************************
+    * @description Returns the default TDTM configuration. If no configuration is currently set, it will return 
+    * the default configuration without setting it. The configuration returned has not yet been inserted in the 
+    * database. It can be manipulated, if necessary, then set. 
+    * If no configuration exists, the application will insert the defaults when any of the triggers run. 
+    * @return List<TDTM_Global_API.TdtmToken> List of trigger handler tokens with the default values. 
+    *********************************************************************************************************/
+    public static List<TDTM_Global_API.TdtmToken> getTdtmConfig() {
+        if(tdtmTokensCache != null) {
+            return tdtmTokensCache;
+        }
+        List<TDTM_Global_API.TdtmToken> tokens = new List<TDTM_Global_API.TdtmToken>();
+        List<Trigger_Handler__c> defaultConfig = TDTM_DefaultConfig.getDefaultRecords();
+        for(Trigger_Handler__c handler : defaultConfig) {
+            tokens.add(new TDTM_Global_API.TdtmToken(handler.Class__c, handler.Object__c, handler.Trigger_Action__c, 
+                    Integer.valueOf(handler.Load_Order__c), handler.Active__c, handler.Asynchronous__c, 
+                    handler.Filter_Field__c, handler.Filter_Value__c, handler.User_Managed__c));
+        }
+        //tdtmTokensCache = tokens; We don't want to set the cache here, but instead have code explicity call setTdtmConfig.
+        return tokens;
+    }
+    
+    /*******************************************************************************************************
+    * @description Sets the trigger handlers configuration with the data passed as parameter.
+    * @parameter tdtmTokens A list with the defaul trigger handlers configuration.
+    * @return void
+    *********************************************************************************************************/
+    public static void setTdtmConfig(List<TDTM_Global_API.TdtmToken> tdtmTokens) {
+        tdtmTokensCache = tdtmTokens;
+        
+        //clear existing TDTM configuration
+        delete [select Id from Trigger_Handler__c limit 50000];
```

Comment:
@davidjray @xinnanlisf I think this is where the TDTM push bug that clears custom TDTM entries was introduced. Back in October...